crates/nostr-mls-storage/src/groups/types.rs

```diff
@@ -0,0 +1,120 @@
+use nostr::{EventId, PublicKey, RelayUrl, Timestamp};
+use serde::{Deserialize, Serialize};
+
+use super::error::GroupError;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum GroupType {
+    /// A group with only two members
+    DirectMessage,
+    /// A group with more than two members
+    Group,
+    /// For handling unknown values
+    Unknown(String),
+}
+
+impl GroupType {
+    /// Safely converts a string to a GroupType, returning an error for invalid values
+    pub fn try_from_string(s: &str) -> Result<Self, GroupError> {
+        match s {
+            "DirectMessage" => Ok(Self::DirectMessage),
+            "Group" => Ok(Self::Group),
+            _ => Err(GroupError::InvalidParameters(format!(
+                "Invalid group type: {}",
+                s
+            ))),
+        }
+    }
+}
+
+impl From<String> for GroupType {
+    fn from(s: String) -> Self {
+        match s.as_str() {
+            "DirectMessage" => Self::DirectMessage,
+            "Group" => Self::Group,
+            _ => Self::Unknown(s),
+        }
+    }
+}
+
+impl From<GroupType> for String {
+    fn from(group_type: GroupType) -> Self {
+        match group_type {
+            GroupType::DirectMessage => "DirectMessage".to_string(),
+            GroupType::Group => "Group".to_string(),
+            GroupType::Unknown(s) => s,
+        }
```

Comment:
I saw that in some parts of code the enums are serialized as lowercase strings. I would keep the same format everywhere. In the others nostr crates I usually write them in lowercase, so we can use the snake case. What do you think? Will this break anything? We can still support deserialization in both formats